docs/design/datacontracts/StackWalk.md

```diff
@@ -0,0 +1,181 @@
+* (3.3) Return the current context and Frame.
+* (3.4) The example has no skipped Frames so we skip this step.
+* (3.5) Since we updated our context and our new context is in managed code, we use the Windows style unwinding tool to iteratively unwind until the context is no longer managed.
+This could take multiple iterations, each time returning a new context to the caller. We end up with context `<C>` which is the first context inside of a native call frame.
+* (3.1) We look at the current Frame, another SoftwareExceptionFrame with attached context `<D>`. Update our working context to `<D>`.
+* (3.2) Since the current Frame is not an InlinedCallFrame, set the current Frame to the next Frame (`NULL TERMINATOR`) on the chain.
+* (3.3) Return the current context and Frame.
+* (3.4) The example has no skipped Frames so we skip this step.
+* (3.5) Again, we updated our context and our new context is in managed code. Therefore, we use the Windows style unwinding tool to iteratively unwind until the context is no longer managed.
+This could take multiple iterations, each time returning a new context to the caller. We end up with context `<E>` which is the first context inside of a native call frame.
+* (3) Since our current context is in native code and the current Frame is the `NULL TERMINATOR`, we are done.
+
+### APIs
+
+The majority of the contract's complexity comes from the stack walking algorithm implementation which is implemented and controlled through the following APIs.
+These handle setting up and iterating the stackwalk state according to the algorithm detailed above.
+
+```csharp
+IStackWalkHandle CreateStackWalk(ThreadData threadData);
+bool Next(IStackWalkHandle stackWalkHandle);
+```
+
+The rest of the APIs convey state about the stack walk at a given point which fall out of the stack walking algorithm relatively simply.
+
+`GetCurrentFrame` creates a copy of the stack walk state which remains constant even if the stack walk is iterated.
+```csharp
+IStackDataFrameHandle GetCurrentFrame(IStackWalkHandle stackWalkHandle);
+```
+
+`GetRawContext` Retrieves the raw Windows thread context of the current frame as a byte array. The size and shape of the context is platform dependent. See [CONTEXT structure](https://learn.microsoft.com/en-us/windows/win32/api/winnt/ns-winnt-context) for more info.
+This context is not guaranteed to be complete. Not all capital "F" Frames store the entire context, some only store the IP/SP/FP. Therefore, at points where the context is based on these Frames it will be incomplete.
+```csharp
+byte[] GetRawContext(IStackDataFrameHandle stackDataFrameHandle);
+```
+
+
+`GetFrameAddress` gets the address of the current capital "F" Frame. This is only valid if the `IStackDataFrameHandle` is at a point where the context is based on a capital "F" Frame. For example, it is not valid when when the current context was created by using the Windows unwinder.
```

Comment:
Same here, I assume we don't use the Windows unwinder API's since this is cross platform, just that our implementation was influenced by the Windows unwinder implementation.